fe/fe-core/src/main/java/com/starrocks/metric/MetricRepo.java

```diff
@@ -526,6 +542,50 @@ private static void collectTableMetrics(MetricVisitor visitor, boolean minifyTab
         }
     }
 
+    private static void collectRoutineLoadProcessMetrics(MetricVisitor visitor) {
+        List<RoutineLoadJob> jobs = Catalog.getCurrentCatalog().getRoutineLoadManager().getRoutineLoadJobByState(
+                Sets.newHashSet(RoutineLoadJob.JobState.NEED_SCHEDULE, RoutineLoadJob.JobState.RUNNING));
+        for (RoutineLoadJob job : jobs) {
```

Comment:
Will it be delay when there are thousands of routine job?

Reply:
Yes, so I add a config param collect_routine_load_lag_metrics.